scripts/quests/outlands/An_Undying_Pledge.lua

```diff
@@ -0,0 +1,132 @@
+-----------------------------------
+-- An Undying Pledge
+-----------------------------------
+-- Log ID: 5, Quest ID: 149
+-- Stray Cloud : !pos -18 1 -20 252
+-- qm5         : !pos 133 -9 221 176
+-----------------------------------
+local ssgID = zones[xi.zone.SEA_SERPENT_GROTTO]
+
+local quest = Quest:new(xi.questLog.OUTLANDS, xi.quest.id.outlands.AN_UNDYING_PLEDGE)
+
+quest.reward =
+{
+    item = xi.item.LIGHT_BUCKLER,
+    fameArea = xi.fameArea.NORG,
+    fame = 50,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.NORG) >= 4
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Stray_Cloud'] = quest:progressEvent(225),
+
+            onEventFinish =
+            {
+                [225] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Stray_Cloud'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
```

Comment:
This charvar should be read into a variable and then used in the condition.

Reply:
What do you mean by this?

Reply:
you mean like local prog = quest:getVar(Blah blah_

if prog == 0 then?

Reply:
Regardless of what goes on behind the scenes, I'll present the following analogy:

Let's say you're doing a research project on whales, and you need to look up the average length of a newborn humpbacked whale.  You have an encyclopedia to look it up, but every time you do, you have to flip through and find the right page.

Instead, what is being suggested is do the research once and write it down.  Xaver beat me to the punch, but I wanted to explain _why_ things like these are important, because its less overhead and a better practice.

Reply:
Ok typically in the past if its been more than 2 id do the do that but now will make it for 2 or more. thanks.